### ISSUE TITLE
php fastcgi_finish_request will increase the CPU consumption

### ISSUE THREAD

**Issue by runtu666** [open-telemetry/opentelemetry-php#1101]
We use fastcgi_finish_request, and the pressure test shows that this will double the CPU consumption. Do you have a good solution?

<img width="1034" alt="image" src="https://github.com/open-telemetry/opentelemetry-php/assets/32669202/55ab9b10-647e-4166-9867-7f4be82b3730">

**Comment by runtu666** [comment]
@brettmc  yes here is the data:
![image](https://github.com/open-telemetry/opentelemetry-php/assets/32669202/4e21a570-1aa8-4e07-a3d6-ce1b031b323b)

**Comment by brettmc** [comment]
Thanks. That doesn't provide much insight, but generally speaking: fastcgi_finish_request is not magic, all of the same processing must happen to export requests, it's just that the client does not wait on it.
During stress testing, we'd need to understand what's going on at a lower level: ending requests earlier may cause your load generator to send more requests, but if all workers are still busy exporting telemetry, they cannot service those requests. That could lead to the CPU usage you see, but I'm only guessing here.

I see that the average response time was slower, as well as the CPU usage being higher. Is that also true at lower volume, when the system is able to keep up? I'd be interested in seeing a graph of response times + resource utilisation as request# increases.

It would be useful if you could also describe the setup of your test. Is it nginx+fpm? Are they on the same host, or different? Which transport are you using, and what backend are you exporting the telemetry to? Where is the backend located? Is it something local and fast, or is it a cloud service? Approximately how many spans does the instrumented code create?

**Comment by brettmc** [comment]
Closing due to lack of response. Please reopen or create a new issue if you have more info!
